### ISSUE TITLE
lint: unlambda false positives

### ISSUE THREAD

**Issue by mastertinner** [go-critic/go-critic#716]
I get the following linting error:

```
unlambda: replace `func(string) error { return errors.New("mocked error") }` with `errors.New`
```

However, this is not true. The replacement would not be the same because it would return the string parameter as an error. My function, however, returns the string `"mocked error"` explicitly as an error.
